src/spec_tools/json_schema.cljc

```diff
@@ -202,7 +202,10 @@
 (defmethod accept-spec 'clojure.spec/and [dispatch spec children]
   (simplify-all-of {:allOf children}))
 
-; merge
+(defmethod accept-spec 'clojure.spec/merge [dispatch spec children]
+  {:type "object"
+   :properties (apply merge (map :properties children))
+   :required (apply concat (map :required children))})
```

Comment:
problem:
`(s/def :animal/says string?)
(s/def :animal/common (s/keys :req [:animal/says]))
(s/def :dog/says boolean?)
(s/def :animal/dog (s/merge :animal/common
                            (s/keys :req [ :dog/says])))`
s/merge work fine for this, but since we only use names :dog/says will overwrite :animal/says.